tests/python/unittest/test_arith_canonical_simplify.py

```diff
@@ -97,6 +97,8 @@ def test_split_index_simplify():
     # cannot simplify mixed case, unless we canonicalize into one mode.
     ck.verify(tdiv(x, 6) * 2 + tmod(fld(x, 3), 2), tdiv(x, 6) * 2 + tmod(fld(x, 3), 2))
 
+    ck.verify(tmod(-x, 2), tmod(x, -2) * -1)
```

Comment:
does this cover modular set analysis?

Reply:
This test only covers the CanonicalSimplify behavior of negative indices, but the ModularSet's behavior should be consistent with the sign convention in CanonicalSimplify.  Since CanonicalSimplify's sign convention for negative indices wasn't checked by any test I could find, I wanted to add a test for its behavior.